### DISCUSSION TITLE
Can pgproto3 package be leveraged?

### DISCUSSION THREAD

**Discussion by kishaningithub** [jeroenrinzema/psql-wire#21] [Ideas]
I see code in this repo where it talks directly in the postgres protocol codes (eg client and server message formats). Do you think the following package by the creator of pgx be leveraged to simplify this codebase?

https://github.com/jackc/pgproto3

**Comment by kishaningithub** [comment]
Just stumbled upon the following by pgx creator which is written on top of above mentioned [pgproto3](https://github.com/jackc/pgmock)

https://github.com/jackc/pgmock

Can this also be leveraged? I find the below code interesting for implementing talking in pg frontend and backend protocols

https://github.com/jackc/pgmock/blob/master/pgmockproxy/proxy/proxy.go

**Reply by jeroenrinzema** [reply]
Thanks for the suggestion! I have used libraries such as `pgproto3` to get a better understanding of the PostgreSQL wire protocol but do not see yet a use case where using such library might be beneficial. Relying on `pgproto3` increases the amount of allocations made per command and restricts us to the features supported inside `pgproto3`. Feel free to let me know if you have some ideas to where `pgproto3` might save us some time.
